lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -136,8 +146,16 @@ void FIRRTLTypesLowering::runOnOperation() {
   ImplicitLocOpBuilder theBuilder(module.getLoc(), &getContext());
   builder = &theBuilder;
 
+  // Remember the original argument attributess.
+  SmallVector<NamedAttribute, 8> originalArgAttrs;
+  DictionaryAttr originalAttrs = module->getAttrDictionary();
+  for (size_t i = 0, e = module.getNumArguments(); i < e; ++i)
+    originalArgAttrs.push_back(
+        originalAttrs.getNamed(getArgAttrName(i)).getValue());
```

Comment:
I think it possible to sink this loop to after the "if argstoremove.empty()" check below, which avoids work in the case where no ports are lowered.